src/ngx_http_lua_uri.c

```diff
@@ -55,8 +60,29 @@ ngx_http_lua_ngx_req_set_uri(lua_State *L)
         return luaL_error(L, "attempt to use zero-length uri");
     }
 
-    if (n == 2) {
+    if (n >= 3) {
+        luaL_checktype(L, 3, LUA_TBOOLEAN);
+        binary = lua_toboolean(L, 3);
+    }
+
+    if (!binary
+        && ngx_http_lua_check_unsafe_uri_bytes(r, p, len, &byte) != NGX_OK)
+    {
+        buf_len = ngx_http_lua_escape_log(NULL, p, len);
+        buf = ngx_palloc(r->pool, buf_len + 1);
+        if (buf == NULL) {
+            return NGX_ERROR;
+        }
+
+        ngx_http_lua_escape_log(buf, p, len);
+        buf[buf_len] = '\0';
+        sprintf(byte_hex, "%02x", byte);
```

Comment:
Why do we need the `byte_hex` buffer in the first place? We can just use `%02x` below in the `luaL_error()` call?